uoyweek.cpp

```diff
@@ -19,7 +34,8 @@ class Term {
         code = c;
         startTS = s;
         endTS = e;
-        //Store actual value instead of reference to gmtime so as to avoid issues
+        /* Store actual value instead of reference to gmtime so as to avoid
+         * issues */
         start = *gmtime(&s);
```

Comment:
dereferencing these looks a bit odd. Maybe a memory leak?

Reply:
`ctime` does some weird stuff with results. If you don't dereference immediately value is lost

Reply:
http://en.cppreference.com/w/cpp/chrono/c/gmtime
> Pointer to a static internal std::tm object on success

No memory leak then, but dereferencing is weird.

Also, below: `(&start)->tm_year`  wut.